doc/user-guide.md

```diff
@@ -400,6 +400,70 @@ func main() {
 
 After adding new import paths to your operator project, run `dep ensure` in the root of your project directory to fulfill these dependencies.
 
+## Leader election
+
+For improved reliability and quick failover it may be necessary to run multiple replicas of the operator with leader election, so that one leader instance handles the reconciliation while the other instances are inactive but ready to take over when the leader fails.
+
+There are two different leader election implementations to choose from, each with its own tradeoff.
+
+- [Leader-for-life][leader_for_life]: The leader pod only gives up leadership (via garbage collection) when it is deleted. Avoids the possibility of 2 instances mistakenly running as leaders(split brain). However, this method can be slow to elect a new leader when the leader pod is on an unresponsive node(partitioned) where it takes a long time for the leader pod to be deleted.
+- [Leader-with-lease][leader_with_lease]: The leader pod periodically renews the leader lease and gives up leadership when it can't renew the lease. This method allows for a faster transition to a new leader when the existing leader is isolated, but there is a possibility of split brain in [certain situations][lease_split_brain].
```

Comment:
I would add that it assumes the leader is capable of stopping what it is doing immediately upon lease expiry. Depending on how a controller is written, it may or may not be able to make that guarantee.

Reply:
@mhrivnak Aren't all leaders capable of stopping abruptly when using leader election via the manager? Since that will just exit the program when the lease expires for the leader and it stops leading.
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/manager/internal.go#L279-L283
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/manager/internal.go#L230-L233
https://github.com/operator-framework/operator-sdk-samples/blob/master/memcached-operator/cmd/manager/main.go#L85-L88

In my understanding this should exit all background goroutines abruptly as well without blocking for any defer statements.

Reply:
Good point, it does look like they've guarded it well. It may be too late to cancel API requests that are in-progress, but that's quite an edge case.